qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/resolvers/IfTruthyValueResolverTest.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.resolvers;
+
+import static com.redhat.qute.QuteAssert.assertHover;
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+import static com.redhat.qute.QuteAssert.testDiagnosticsFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for 'ifTruthy' value resolver.
+ *
+ * @see https://quarkus.io/guides/qute-reference#built-in-resolvers
+ *
+ */
+public class IfTruthyValueResolverTest {
+
+	@Test
+	public void diagnostics() {
+		String template = "{@org.acme.Item item}\r\n" + //
+				"{item.ifTruthy}";
+		testDiagnosticsFor(template);
+
+		template = "{@org.acme.Item item}\r\n" + //
+				" \r\n" + //
+				"{item.name.ifTruthy(item.name)}";
+		testDiagnosticsFor(template);
+	}
+
+	@Test
+	public void completion() throws Exception {
+		String template = "{@org.acme.Item item}\r\n" + //
+				"{item.name.|}";
+		testCompletionFor(template, //
+				c("ifTruthy(base : T, arg : java.lang.Object) : T", "ifTruthy", r(1, 10, 1, 10)));
+		template = "{@org.acme.Item item}\r\n" + //
+				"{item.name.ifTruthy('abcd').|}";
+		testCompletionFor(template, //
+				c("name : String", "name", r(1, 27, 1, 27)), //
+				c("price : BigInteger", "price", r(1, 27, 1, 27)), //
+				c("review : Review", "review", r(1, 27, 1, 27)), //
+				c("review2 : Review", "review2", r(1, 27, 1, 27)), //
+				c("getReview2() : Review", "getReview2", r(1, 27, 1, 27)));
+	}
+
+	@Test
+	public void hover() throws Exception {
+		String template = "{@org.acme.Item item}\r\n" + //
+				" \r\n" + //
+				"{item.name.i|fTruthy(item)}";
+		assertHover(template, "```java" + //
+				System.lineSeparator() + //
+				"T ifTruthy(Object arg)" + //
+				System.lineSeparator() + //
+				"```" + //
+				System.lineSeparator() + //
+				"Outputs the default value if the previous part cannot be resolved or resolves to null." + //
```

Comment:
"Outputs the default value if the base object cannot be resolved or the base Object otherwise" + //